actions/checkForAndPlaceOrder.ts

```diff
@@ -119,91 +167,80 @@ const _checkForAndPlaceOrder: ActionFn = async (
     }
   }
 
+  // Delete owners with no orders
+  for (const [owner, conditionalOrders] of Array.from(ownerOrders.entries())) {
+    if (conditionalOrders.size === 0) {
+      ownerOrders.delete(owner);
+    }
+  }
+
```

Comment:
This is quite valid, and most likely exactly what we want to do. Another thing is that having the keys for the `ownerOrders` (ie. previous safes that have traded), allows us to possibly use this as some cache for valid safes that are likely to have upgraded their fallback handler etc. If you can think of a use case for this, it may be worthwhile keeping around the keys, otherwise we can bin them.

Reply:
I will comment out for now, and we decide later

I think one use case, is just to avoid doing the verification. I just added this to not keep adding owners and big loops over time. I would hope, once the order is executed or expired, we don't need to keep things around. 

Leaving it for now